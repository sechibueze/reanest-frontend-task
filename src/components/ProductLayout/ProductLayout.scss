@import "../../assets//theme/colors.scss";
.product-layout-wrapper {
  display: none;
  grid-template-columns: repeat(1fr, 2);
  row-gap: 1rem;
  padding: 1rem 1.5rem;
  background-color: #f3eaea;
  border-radius: 0.25rem;
  box-shadow: 0px -1px 4px $greyed-out;
  @media screen and (min-width: $mobile-breakpoint) {
    & {
      display: grid;
    }
  }
}
$product-item-border-rad: 0.7rem;
.product-item {
  display: grid;
  grid-template-columns: 2rem auto;
  grid-template-rows: repeat(1fr, 3rem);
  column-gap: 1rem;
  background-color: $white;
  border-radius: $product-item-border-rad;
  height: 3rem;
}
.product-data {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;

  font-size: 0.8rrem;
}
.product-image {
  text-align: left;
  border-top-left-radius: $product-item-border-rad;
  border-bottom-left-radius: $product-item-border-rad;

  width: 3rem;
  height: 100%;
}
.product-name {
  padding: 1rem 1rem;
  font-size: 0.8rem;
}

.product-item-meta {
  flex: 1;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  font-size: 0.6rem;
  color: gray;
  .product-category,
  .product-date {
    padding: 0 1rem;
  }
}

.product-status {
  padding: 0.25rem;
  border-radius: 0.4rem;
  background-color: $blue;
  color: $white;
  font-size: 0.5rem;
  font-weight: bold;
  text-align: center;
}
