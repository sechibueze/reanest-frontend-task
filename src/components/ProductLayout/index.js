import productImage from "../../assets/images/icon.jpg";
import { testData } from "../../_data";
import "./ProductLayout.scss";
const ProductLayout = ({ transactions }) => {
  return (
    <div className="product-layout-wrapper">
      {transactions.items.map((item, idx) => {
        const { image, title, category, date, status } = item;
        return (
          <div className="product-item" key={idx}>
            <img
              src={image || productImage}
              alt="product visuals"
              className="product-image"
            />
            <div className="product-data">
              {/* <div className="product-item-alpha"> */}
              <p className="product-name"> {title && title} </p>
              {/* </div> */}
              <div className="product-item-meta">
                <span className="product-category"> {category} </span>
                <span className="product-date"> {date} </span>
                <span className="product-status"> {status} </span>
                <span className="product-action">
                  {" "}
                  <input type="checkbox" />{" "}
                </span>
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );
};
ProductLayout.defaultProps = {
  transactions: testData.transactions,
};
export default ProductLayout;
