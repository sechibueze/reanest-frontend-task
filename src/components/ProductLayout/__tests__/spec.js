import { shallow } from "enzyme";
import ProductLayout from "../index";
const productCount = 3;
describe("<Navbar /> renders correctly", () => {
  const component = shallow(<ProductLayout />);
  const productLayoutWrapper = component.find(".product-layout-wrapper");
  const productItems = component.find(".product-item");
  it("ProductLayout matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("<ProductLayout /> was rendered correctly", () => {
    expect(productLayoutWrapper.length).toEqual(1);
  });
  it("Product items rendered correctly", () => {
    expect(productItems.length).toEqual(productCount);
  });
});
