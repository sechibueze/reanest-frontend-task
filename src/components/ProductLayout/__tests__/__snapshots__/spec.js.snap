// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly Navbar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`<Navbar /> renders correctly ProductLayout matches snapshot 1`] = `ShallowWrapper {}`;
